syntax = "proto3";

package tesla.proto.energy_device.v1;

option go_package = "github.com/teslamotors/energy_device/pkg/protocol/protobuf/energydevice/v1";
option java_package = "com.tesla.generated.energy_device.v1";
option java_outer_classname = "MessageEnvelope";

import "tesla_api/protobuf/energy_device/v1/common_messages.proto";
import "tesla_api/protobuf/energy_device/v1/delivery_channel.proto";
import "tesla_api/protobuf/energy_device/v1/participant.proto";
import "tesla_api/protobuf/energy_device/v1/teg_messages.proto";

message MessageEnvelope {
  DeliveryChannel delivery_channel       = 1;
  Participant sender                     = 2;
  Participant recipient                  = 3;
  reserved 7;
  oneof payload {
    CommonMessages common                = 4;
    TEGMessages teg                      = 5;
    //WCMessages wc                       = 6;
    //NeurioMeterMessages neuriometer     = 9;
    //EnergySiteNetMessages energysitenet = 10;
    //AuthorizationMessages authorization = 12;
    //FileStoreMessages filestore         = 15;
    //GraphQLMessages graphql             = 16;
  }
}